### PR TITLE
Feature/add dump tab

### PR DESCRIPTION
**Pull request by phobos2390**
Added dump tab viewer. Creates tab with dump printed. The Tab is annotated, enabling important data to be highlighted.

Additionally, the context menu for the picture viewer tab has an option to select the image header and raw data from within the dump.

### CODE REVIEW

File: OpenLocoTool/Annotation.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		public Annotation(string name, int start, int length)
+		{
+			Name = name;
+			this.start = start;
+			Length = length;
+		}
+
+		public string Name { get => name; set => name = value; }
+		public Annotation? Parent { get => parent; set => parent = value; }
+		public int Start { get => start; set { start = value; Length = Length; } }
```
Comment (LeftofZen) [inline OpenLocoTool/Annotation.cs:26]: bug here, should be `Length = length`, and fwiw you can use autoproperties for all this or even records, it'll save having to write so much code here

Comment (LeftofZen) [reply OpenLocoTool/Annotation.cs:26]: actually maybe not

**Review by LeftofZen — APPROVED** [review]

File: OpenLocoTool/Annotation.cs (hunk 1 above)
Comment (phobos2390) [reply OpenLocoTool/Annotation.cs:26]: Changed to Length = length. Part of the idea is that setting Start to a new value keeps the length the same and recalculates the End value. Not necessarily that important.

File: OpenLocoTool/DatFileParsing/LocoObject.cs [hunk 2]
```diff
@@ -59,6 +59,8 @@ public interface ILocoStructVariableData
 	{
 		ReadOnlySpan<byte> Load(ReadOnlySpan<byte> remainingData);
 
+		//int Annotate(ReadOnlySpan<byte> remainingData, int running_count, IDictionary<int,string> annotations);
```
[the remark this answers is not in the document]
Comment (phobos2390) [reply OpenLocoTool/DatFileParsing/LocoObject.cs:62]: Removed. Haven't added detailed variable data annotations per object but that could be something to add as bits of deserialization logic when the specific object gets resolved

**Comment by phobos2390** [comment]
Created a new pull request: https://github.com/LeftofZen/OpenLocoTool/pull/8

**Pull request merged**